json-dns/unmarshal.go

```diff
@@ -38,7 +38,7 @@ func PrepareReply(req *dns.Msg) *dns.Msg {
 	reply := new(dns.Msg)
 	reply.Id = req.Id
 	reply.Response = true
-	reply.Opcode = reply.Opcode
+	reply.Opcode = req.Opcode
 	reply.RecursionDesired = req.RecursionDesired
```

Comment:
Wow thank you for this fix!
I have never realized this bug!

Reply:
Randomly found thanks to IDE's syntax highlighting :) it's something picked by the go linter AFAIK.